### DISCUSSION TITLE
Rtsp_simple_server on HTML5 player

### DISCUSSION THREAD

**Discussion by kentusch** [bluenviron/mediamtx#345] [General]
Stream to html5 player from ubuntu in a virtual machine
Hello, I cannot understand how I can change ip4 127.0.0.1 in SDP, the structure is like this:
Docker on a virtual machine (Ubuntu) distributes a stream via rtsp tsp to an external machine where there is a player with a proxy (that is, another external one) Ubuntu-> Windows-> Player (proxy) -> 127.0.0.1 (Windows)
 and when the SDP tells him to connect to 127.0.0.1, he turns to the host machine, how can I specify the ip in the SDP so that it addresses me to a virtual one?
I would not ask questions, but it seems to me that this can not be fixed anywhere except the rtsp_simple_ server itself.
If you need any information, tell me,please.
![h264](https://user-images.githubusercontent.com/76987642/114052683-73415d00-98b8-11eb-8536-cc6d4c017d17.png)

Thank you!

**Comment by aler9** [comment]
Hello, the problem here isn't the "127.0.0.1" ip, the problem here is that Streamedian's html5_rtsp_player uses this URL to setup the track

```
rtsp://192.168.1.20:8554/livestream//trackID=0
```

while it should use

```
rtsp://192.168.1.20:8554/livestream/trackID=0
```

I hacked the server to support these kind of URLs, please try this nightly release and let me know if it works
[rtsp-simple-server_v0.15.4-3-g434f416_darwin_amd64.tar.gz](https://github.com/aler9/rtsp-simple-server/files/6280471/rtsp-simple-server_v0.15.4-3-g434f416_darwin_amd64.tar.gz)
[rtsp-simple-server_v0.15.4-3-g434f416_linux_amd64.tar.gz](https://github.com/aler9/rtsp-simple-server/files/6280473/rtsp-simple-server_v0.15.4-3-g434f416_linux_amd64.tar.gz)
[rtsp-simple-server_v0.15.4-3-g434f416_linux_arm6.tar.gz](https://github.com/aler9/rtsp-simple-server/files/6280475/rtsp-simple-server_v0.15.4-3-g434f416_linux_arm6.tar.gz)
[rtsp-simple-server_v0.15.4-3-g434f416_linux_arm7.tar.gz](https://github.com/aler9/rtsp-simple-server/files/6280476/rtsp-simple-server_v0.15.4-3-g434f416_linux_arm7.tar.gz)
[rtsp-simple-server_v0.15.4-3-g434f416_linux_arm64v8.tar.gz](https://github.com/aler9/rtsp-simple-server/files/6280477/rtsp-simple-server_v0.15.4-3-g434f416_linux_arm64v8.tar.gz)
[rtsp-simple-server_v0.15.4-3-g434f416_windows_amd64.zip](https://github.com/aler9/rtsp-simple-server/files/6280478/rtsp-simple-server_v0.15.4-3-g434f416_windows_amd64.zip)

**Reply by kentusch** [reply]
Thank you very much for your answer, I did not use these versions, but changed the player's code, thanks to which the source was successfully reproduced.
I haven't tested the versions, but if you like, with some free time I could do it.

[a comment by an automated account is not shown]
